jena-arq/src/test/java/org/apache/jena/sparql/algebra/TestOpAsQuery.java

```diff
@@ -482,6 +482,21 @@ public void testMinus02() {
                                                            "SELECT * { GRAPH ?g { { ?x ?y ?z FILTER(EXISTS { ?s ?p ?o }) } ?x ?y ?z } }",
                                                               syntaxARQ); }
 
+    @Test
```

Comment:
These EXISTS tests are about the ARQ extensions.

IIRC It's not about EXISTS, it's about subquery select expressions.

If that's right please use this test (insert at around L463):

```java
  @Test
  public void testSubQuery01() {
      test_roundTripQuery("SELECT ?z { SELECT ?x {  } }");
  }
```

The outer `SELECT ?z` stops the outer `SELECT` getting removed because `SELECT *` is a no-op.

Reply:
Your query does not fail for me.

The fix is in 
```
    public Expr transform(ExprFunctionOp funcOp, ExprList args, Op opArg)
    {
        Element el2 = ElementTransformer.transform(funcOp.getElement(), transform, this);
    }
```
So the issue is related to (NOT) EXISTS *within* in a subquery.
Also note that, the transform is an `ElementTransform` - so `SELECT * { ?s a ?c }` is different from `?s a ?c` - although the Op would be the same.

I reduced the query to this:
```
SELECT ?s {
  SELECT ?s {
    ?s a ?t
    FILTER EXISTS { SELECT * { ?s a ?c } }
  }
}
```